rpy2/ipython/rmagic.py

```diff
@@ -169,6 +170,50 @@ def py2rpy_list(obj):
     return res_rpy
 
 
+def _find(name: str, ns: dict):
+    """Find a Python name, which might include dot-separated path to a name.
+
+    Args:
+    - name: a key in dict if no dot ('.') in it, otherwise
+    and sequence of dot-separated namespaces with the name of the
+    object last (e.g., `package.module.name`).
+    Returns:
+    The object wanted. Raises a NameError or an Attribute error if not found.
+    """
+
+    obj = None
+    obj_path = name.split('.')
+    look_for_i = 0
+    try:
+        obj = ns[obj_path[look_for_i]]
+    except KeyError:
+        raise NameError(
+            f"name '{obj_path[look_for_i]}' is not defined."
+        )
+    look_for_i += 1
+    while look_for_i < len(obj_path):
+        try:
+            obj = getattr(obj, obj_path[look_for_i])
+        except AttributeError:
+            raise AttributeError(
+                f"'{'.'.join(obj_path[:look_for_i])}' "
+                f"has no attribute '{obj_path[look_for_i]}'."
+            )
```

Comment:
```suggestion
        except AttributeError as e:
            raise AttributeError(
                f"'{'.'.join(obj_path[:look_for_i])}' "
                f"has no attribute '{obj_path[look_for_i]}'."
            ) from e
```

for nicer traceback?

Reply:
Yes. Thanks.